client/src/components/LoginForm.js

```diff
@@ -52,6 +54,8 @@ export default function LoginForm() {
                                 setSeverity("success");
                                 setMessage("Successfully logged in!");
                                 setOpen(true);
+                                user.setProfile(data.user)
```

Comment:
Generally we'll only want setProfile called from the userContext. I think we should move this line into the login function, and move the login function into userContext.

Reply:
I think I understand but want to double check on one part.  If I move the setProfile and login function to the userContext, won't I still need to pass the LoginForm and SignUp a function from the UseContextProvider so that those components can pass the form values to the login funcitons?

Reply:
All understood and I have refactored to have the fetch requests inside the UseContextProvider as well as added in useReducer to help with updating state in the future.